### PR TITLE
JBrowse-next (0.7.0)

### LINKED ISSUE #993: JBROWSE_SOURCE_DIR is not defined when installing Jbrowse
The environment variable JBROWSE_SOURCE_DIR is not defined when installing Jbrowse. When executing jbrowse in Galaxy no directory is passed to the script leading to the following error:

```
Fatal error: Exit code 1 ()
usage: jbrowse.py [-h] [--jbrowse JBROWSE] [--outdir OUTDIR] [--standalone]
                  xml
jbrowse.py: error: argument --jbrowse: expected one argument
mv: cannot stat `/remote/RMS/users/galaxyd/job_working_directory/008/8727/dataset_50589_files/index.html': No such file or directory
```

### PR DESCRIPTION
**Pull request by erasche**
Did a really messy rebase so this may need a bit extra testing :(

Switch to conda fixes #993

### CODE REVIEW

File: tools/jbrowse/macros.xml [hunk 1]
```diff
@@ -180,7 +225,7 @@ This Galaxy tool relies on the JBrowse, maintained by the GMOD Community. The Ga
             <param type="select" label="JBrowse style.color function's score scaling" name="score_scaling"
                    help="How should the colors be distributed across the values? For blast results which distributes scores on the scale of approximately [1e-500, 10], it makes sense to request a logarithmic scaling of the color values. Logarithmic is indeed the default for blast. However other analysis methods may produce scores on ranges such as [0, 100] where a linear scale would be more appropriate for color distribution.">
               <option value="linear"      selected="@SCALING_LIN_SELECT@" >Linear scaling</option>
-              <option value="logarithmic">Logarithmic scaling</option>
+              <option value="logarithmic" selected="false"                >Logarithmic scaling</option>
```
Comment (abretaud) [inline tools/jbrowse/macros.xml]: There are a few extra spaces

File: tools/jbrowse/jbrowse.py [hunk 2]
```diff
@@ -82,11 +84,11 @@ class ColorScaling(object):
         """,
         'blast': """
             var opacity = 0;
-            if(score == 0.0) {
+            if(score == 0.0) {{
```
Comment (abretaud) [inline tools/jbrowse/jbrowse.py]: Any reason to change this? (same thing at l67)

Comment (erasche) [inline tools/jbrowse/jbrowse.py]: yes, they were getting eaten up by the templating language. It interpreted { opacity=1 } as a variable name which was very wrong. Same with all of the other braces in the templated JS, it was a big bug I should've caught earlier.

Comment (abretaud) [inline tools/jbrowse/jbrowse.py]: Ah ok, I hadn't see it was a template!

**Comment by abretaud** [comment]
By the way, while I think about it, I had a problem a few days ago on the jbrowse track names: they are a little too much sanitized, e.g. single quotes are replaced by __sq__
I'm not sure if we can act on this as it comes from element_identifiers

**Comment by erasche** [comment]
as `__sq__`? If so easy to replace.

**Comment by bgruening** [comment]
We would like to have some more real-user testing here :)

**Comment by erasche** [comment]
I need to do more testing on the ComboTrackSelector but it is getting close to done.

**Comment by abretaud** [comment]
Maybe next week I'll have some time to make some tests

**Comment by erasche** [comment]
I was thinking more "real-life testing" but yes, tests for that would be awesome as well! I'm mostly worried about getting the galaxy URL from the infrastructure_url OK, so we can link back to the datasets/tools.

**Comment by abretaud** [comment]
I was thinking about manual tests with real data :)

**Comment by erasche** [comment]
Oh! Hah, that works too.

**Comment by erasche** [comment]
> had to set GALAXY_CONFIG_GALAXY_INFRASTRUCTURE_URL

did it crash without it? I'd like it to not fail too miserably.

**Comment by abretaud** [comment]
No crash of the tool, it's just that the links to datasets were wrong in jbrowse, they were looking like `href="localhost/datasets/blablabla"` which the browser interprets as a link relative to jbrowse url => 404 error

**Comment by erasche** [comment]
Ah, gotcha, thanks!

**Comment by erasche** [comment]
ok, added the help text / forced protocol-less URLs to have a protocol (`http://`, should be OK.)

**Comment by erasche** [comment]
should be ready for final testing / merging!

**Comment by abretaud** [comment]
Great! Can you merge master into this to fix the flake8 errors (https://github.com/galaxyproject/tools-iuc/commit/a5d7d394c7522aaebdc9579c2f32f8b98d37820f)?

**Comment by abretaud** [comment]
Green!
I took the liberty to make a last rebase as I thought it would take me 2 minutes.. It turned out I had to fight with travis which didn't like my force push.
Unless you have some last minute changes to do, I'll merge it soon
Can't wait to use it!

**Comment by erasche** [comment]
Nope, I'm happy. Merged when you're ready! :D

**Comment by bgruening** [comment]
Wuhu, what happend here! Awesome!

**Pull request merged**